x/wasm/internal/keeper/keeper.go

```diff
@@ -263,7 +263,7 @@ func (k Keeper) Migrate(ctx sdk.Context, contractAddress sdk.AccAddress, caller
 	return types.ResultFromData(res.Data), nil
 }
 
-// UpdateContractAdmin sets the admin value on the ContractInfo. New admin can be nil to disable further migrations/ updates.
+// UpdateContractAdmin sets the admin value on the ContractInfo. It must be a valid address (use ClearContractAdmin to remove it)
 func (k Keeper) UpdateContractAdmin(ctx sdk.Context, contractAddress sdk.AccAddress, caller sdk.AccAddress, newAdmin sdk.AccAddress) error {
```

Comment:
If that code worked for clearing before, don't we need an additional check now?

Reply:
This is checked in `ValidateBasic`: https://github.com/CosmWasm/wasmd/pull/157/files#diff-47cb7ef32f0dc71704a9c4ac79c93817L252

I can check again, but that is guaranteed to be called before processing a message and is the standard way to enforce such guarantees (same with enforcing the signers). I personally dislike the magic/non-transparency but this is the cosmos-sdk-style

Reply:
Alright, thanks for clarification. Makes sense how it is.

The whole idea of struct validation is questionable. A strict type should never allow instantiating invalid data. Then you only have one way to create that type and this does all the checks. But this is a discussion for some other place and time.